### ISSUE TITLE
 'Form Tags' Idea

### ISSUE THREAD

**Issue by ab2d3e8** [bootstrap-vue/bootstrap-vue#4747]
when use Object for 'Form Tags', There is not have text and value.
example  options: ['Apple', 'Orange', 'Banana', 'Lime', 'Peach', 'Chocolate', 'Strawberry'],
when I change to :
options: [{ text:  'Apple',value:'A1'}, { text:  'Orange',value:'A2'}, { text:  'Banana',value:'A3'}, { text:  'Lime',value:'A4}, { text:  'Peach',value:'A5}, { text:  'Chocolate',value:'A4}, { text:  'Strawberry',value:'A4}]
Is not visible text. 

Is possible to use value like key.

**Comment by ab2d3e8** [comment]
I found solution like:
```
<b-form-tag
	:disabled="disabled"
	:title="tag"
	class="ml-1"
	variant="primary"
	@remove="removeTag(tag)"
>
{{  find_Text_with_Value(tag) }}
</b-form-tag>
```
find_Text_with_Value is methods.

**Comment by tmorehouse** [comment]
Since tags allow (by default) any arbitrary text to to be entered by the user (as tags are meant to be short strings by nature), it would be difficult to handle the model being an array of objects.

**Comment by jackmu95** [comment]
@tmorehouse I personally see that as a useful feature too.

We could normalize the input from the `addTag()` method to an object (e.g. `'New Tag'` to `{ value: 'New Tag' }`). This would also give us the possibility to apply per-tag props like `variant`.

**Comment by tmorehouse** [comment]
It would have to be controlled by a new prop to enable returning an array of objects instead of an array of strings

**Comment by jackmu95** [comment]
@tmorehouse Yeah, definitely.

**Comment by jackmu95** [comment]
Closes in favor of #4617.
